### ISSUE TITLE
Error while trying to access ML Studio from the UI

### ISSUE THREAD

**Issue by dosiennik** [awslabs/aws-dataall#47]
### Describe the bug

I am getting the following error when try to click on the ML Studio link in the data.all UI:

![sagemakerstudio_ bug](https://user-images.githubusercontent.com/30467028/173605115-a12f9a02-35b6-45d7-8732-1377f1645640.png)

### How to Reproduce

1. Setup ML Studio in the data.all UI
2. Try to access it by clicking on the ML Studio Name

### OS

mac

### Python version

3.8

### AWS data.all version

*

**Comment by dosiennik** [comment]
Thanks, I moved a bit forward with the mentioned fix although it still doesn't work for me. It ends up with the following error when I am trying to access SageMaker generated pre-signed url:

`SageMaker is unable to use your associated ExecutionRole [arn:aws:iam:::role/dataall-int-pihvzcf8] to create app. Verify that your associated ExecutionRole has permission for 'sagemaker:CreateApp'.`

Looks like the role assigned to the SageMaker UserProfile doesn't have a privilege to create an app although I see the following in its policy:

`{
            "Condition": {
                "StringEquals": {
                    "aws:RequestTag/Team": [
                        "DAAdministrators"
                    ]
                }
            },
            "Action": "databrew:Create*",
            "Resource": "*",
            "Effect": "Allow"
        },`
        
        When I add the following createApp privilege explicitly to the role:
        `{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "sagemaker:CreateApp",
            "Resource": "*"
        }
    ]
}`

Then it works. Not sure what is the root cause, is it related somehow to the tagging? Shouldn't the tags from the UserProfile be propagated to the application?

**Comment by LEUNGUU** [comment]
@dosiennik Hello. Thanks for your reply. I dig into it and below is my findings.
- The meaning of below policy is to allow request to create sagemaker related resource **only when the request contains the tag {Team: DAAdministrators}**. But what we use now is [create_presigned_domain_url](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_presigned_domain_url), which will launch a default app without the required tags. That is why we have the policy but it doesn't work.
 ```yaml
{
            "Condition": {
                "StringEquals": {
                    "aws:RequestTag/Team": [
                        "DAAdministrators"
                    ]
                }
            },
            "Action": "sagemaker:Create*",
            "Resource": "*",
            "Effect": "Allow"
        }
```

- I also check another one called **aws:ResourceTag**. This determine whether to allow access to the resource based on the tags that are attached to the resource. So I changed the _ RequestTag_ to _ResourceTag_, but it still doesn't work. I think this is normal because we cannot control the access to what we don't have now.

So I propose a simple solution that  we can remove the condition to make it work for now. Later when we add the creation of sagemaker domain we can orchestrate all these things, such as domain, user profile and app.

**Comment by dosiennik** [comment]
@LEUNGUU thanks a lot for checking and replying. Yeah this will fix it since it works when I did such a change manually for testing.

It is actually odd considering UserProfile's tags definition:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-userprofile.html

Especially at the part: 'Tags that you specify for the User Profile are also added to all apps that the User Profile launches.'

I checked the tags on the user profile, they are there as expected but seems like they are not propagated to the SageMaker app.

Anyway I agree that the proposed fix will solve the issue although not fully sure about the root cause.
